---
title: C28198
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28198
helpviewer_keywords:
- C28198
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
ms.openlocfilehash: 10236f359bff722c0b63bb0e9db7a7f38d41b7e8
ms.sourcegitcommit: 7bea0420d0e476287641edeb33a9d5689a98cb98
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 02/17/2020
ms.locfileid: "79466181"
---
# <a name="c28198"></a>C28198

> ADVERTENCIA C28198: posiblemente pérdida de memoria debido a una excepción.

Esta advertencia indica que no se libera memoria asignada después de que se produzca una excepción. La instrucción al final de la ruta de acceso puede producir una excepción. La memoria se pasó a una función que podría haber guardado una copia que se va a liberar más adelante.

Esta advertencia es muy similar a la advertencia [C28197](../code-quality/c28197.md). Las anotaciones que se recomiendan para su uso con [C28197](../code-quality/c28197.md) de advertencia también se pueden usar aquí.

## <a name="example"></a>Ejemplo

El siguiente código de ejemplo genera esta advertencia:

```cpp
char *p1 = new char[10];
char *p2 = new char[10];

test(p1);   // does not save a copy of p

delete[] p2;
delete[] p1;
```

El siguiente código de ejemplo evita esta advertencia:

```cpp
char *p1 = new char[10];
char *p2 = NULL;

test(p1);   // does not save a copy of p
try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
```
